src/ADAL.PCL.Android/BrokerProxy.cs

```diff
@@ -0,0 +1,663 @@
+                }
+            }
+            catch (OperationCanceledException e)
+            {
+                PlatformPlugin.Logger.Error(null, e);
+            }
+            catch (Exception e)
+            {
+                // Authenticator gets problem from webrequest or file read/write
+                PlatformPlugin.Logger.Error(null, new AdalException("Authenticator cancels the request", e));
+            }
+
+            return intent;
+        }
+
+        private string GetRedirectUriForBroker()
+        {
+            string packageName = Application.Context.PackageName;
+
+            // First available signature. Applications can be signed with multiple
+            // signatures.
+            string signatureDigest = this.GetCurrentSignatureForPackage(packageName);
+            if (!String.IsNullOrEmpty(signatureDigest))
+            {
+                return String.Format("{0}://{1}/{2}", RedirectUriScheme, EncodingHelper.UrlEncode(packageName), EncodingHelper.UrlEncode(signatureDigest));
+            }
+
+            return String.Empty;
+        }
+
+        private string GetCurrentSignatureForPackage(string packageName)
+        {
+            try
+            {
+                PackageInfo info = Application.Context.PackageManager.GetPackageInfo(packageName, PackageInfoFlags.Signatures);
+                if (info != null && info.Signatures != null && info.Signatures.Count > 0)
+                {
+                    Signature signature = info.Signatures[0];
+                    MessageDigest md = MessageDigest.GetInstance("SHA");
+                    md.Update(signature.ToByteArray());
+                    return Convert.ToBase64String(md.Digest(), Base64FormattingOptions.None);
+                    // Server side needs to register all other tags. ADAL will
+                    // send one of them.
+                }
+            }
+            catch (PackageManager.NameNotFoundException)
+            {
+                PlatformPlugin.Logger.Information(null, "Calling App's package does not exist in PackageManager");
+            }
+            catch (NoSuchAlgorithmException)
+            {
+                PlatformPlugin.Logger.Information(null, "Digest SHA algorithm does not exists");
+            }
+
+            return null;
+        }
+
+        private Bundle GetBrokerOptions(AuthenticationRequest request)
+        {
+            Bundle brokerOptions = new Bundle();
+            // request needs to be parcelable to send across process
+            brokerOptions.PutInt("com.microsoft.aad.adal:RequestId", request.RequestId);
+            brokerOptions.PutString(BrokerConstants.ACCOUNT_AUTHORITY,
+                    request.Authority);
+            brokerOptions.PutInt("json", 1);
+            brokerOptions.PutString(BrokerConstants.ACCOUNT_RESOURCE,
+                    request.Resource);
+            string s = GetRedirectUriForBroker();
```

Comment:
this is the current behavior that I am mimicing from Android. If you have not changed this then we should discuss in the scrum.